### PR TITLE
implement multiplayer launcher (based on Chocolate Doom)

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
This PR is a bit frightening as it is. Could you split the refactoring out into a separate PR so we can get this right before attempting to merge the multiplayer launcher?

File: Source/i_main.c [hunk 1]
```diff
@@ -29,28 +29,57 @@
...
+
+// Descriptions taken from MSDN
+static struct {
+  int sig;
+  const char *description;
+} sig_desc[] = {
+  { SIGABRT, "Abnormal termination" },
+  { SIGFPE,  "Floating-point error" },
+  { SIGILL,  "Illegal instruction" },
+  { SIGINT,  "CTRL+C signal" },
```
Comment (fabiangreffrath) [inline Source/i_main.c:49]: "Interrupt signal"?

File: Source/m_menu.c [hunk 2]
```diff
@@ -5198,10 +5198,11 @@ boolean M_Responder (event_t* ev)
 
         if (M_InputActivated(input_demo_fforward))
         {
-          if (demoplayback && singledemo && !demo_skipping)
+          if (demoplayback && singledemo && !demo_skipping && !fastdemo)
           {
-            fastdemo = !fastdemo;
-            I_SetFastdemoTimer();
+            static boolean on = false;
+            on = !on;
```
Comment (fabiangreffrath) [inline Source/m_menu.c:5204]: This is confusing.

Comment (fabiangreffrath) [reply Source/m_menu.c:5205]: I specifically mean the `on = !on` line. Could you rename the variable to something different, e.g. `fast`?

File: Source/CMakeLists.txt [hunk 3]
```diff
@@ -189,6 +190,39 @@ target_compile_definitions(woof PRIVATE MINIZ_NO_STDIO)
...
+    d_iwad.c                d_iwad.h
+    net_io.c                net_io.h
+    net_packet.c            net_packet.h
+    net_sdl.c               net_sdl.h
+    net_query.c             net_query.h
+    net_structrw.c          net_structrw.h
+    z_zone.c                z_zone.h
+    ../win32/win_fopen.c    ../win32/win_fopen.h)
+
+add_executable(multiplayer WIN32 ${SETUP_SOURCES})
```
[the remark this answers is not in the document]
Comment (rfomin) [reply Source/CMakeLists.txt:209]: Ok, I rename it.

File: Source/i_main.c (hunk 1 above)
Comment (fabiangreffrath) [reply Source/i_main.c:49]: Well, okay. 😉

**Comment by rfomin** [comment]
> Setup stuff doesn't necessarily have to be in a separate subdirectory, if you are asking me.

I'll try to copy Choco's approach, I think it's good.

**Pull request closed without merging**